tutorial/versioning/version_ranges.rst

```diff
@@ -0,0 +1,147 @@
+    ...
+    Requirements
+        pkg/1.0
+
+Then, if we create a new version of ``pkg/1.1``, it will not automatically be used by ``app``:
+
+.. code-block:: bash
+
+    $ conan create pkg --version=1.1
+    ... pkg/1.0 ...
+    # Note how this still uses the previous 1.0 version
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.0
+
+So we could modify ``app`` conanfile to explictly use the new ``pkg/1.1`` version, but instead of that,
+let's use the following version-range expression (introduced by the ``[expression]`` brackets):
+
+.. code-block:: python
+    :caption: app/conanfile.py
+
+    from conan import ConanFile
+
+    class pkgRecipe(ConanFile):
+        name = "app"
+        requires = "pkg/[>=1.0 <2.0]"
+
+When we now install the dependencies of ``app``, it will automatically use the latest version in the
+range, even if we create a new one, without needing to modify the ``app`` conanfile:
+
+.. code-block:: bash
+
+    # this will now use the newer 1.1
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.1
+
+    $ conan create pkg --version=1.2
+    ... pkg/1.2 ...
+    # Now it will automatically use the newest 1.2
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.2
+
+This holds as long as the newer version lies within the defined range, if we create a ``pkg/2.0`` version,
+``app`` will not use it:
+
+.. code-block:: bash
+    
+    $ conan create pkg --version=2.0
+    ... pkg/2.0 ...
+    # Conan will use the latest in the range
+    $ conan install app
+    ...
+    Requirements
+        pkg/1.2
+
+
+Semantic versioning
+-------------------
+
+The semantic versioning specification or `semver <https://semver.org/>`_, specifies that packages should
+be versioned using always 3 dot-separated digits like ``MAJOR.MINOR.PATCH``, with very specific meanings for each digit.
+
+Conan extends the semver specification to any number of digits, and also allows to include letters in it.
+This was done because during 1.X a lot of experience and feedback from users was gathered, and it became evident
+than in C++ the versioning scheme is often more complex, and users were demanding more flexibility, allowing
+versions like ``1.2.3.a.8`` if necessary.
+
+The ordering of versions when necessary (for example to decide which is the latest version in a version range)
+is done by comparing individually each dot-separated entity in the version, from left to right. Digits will be
+compared numerically, so 2 < 11, and entries containing letters will be compared alphabetically (even if they
+also contain some numbers).
+
+Similarly to the semver specification, Conan can manage **prereleases** and **builds** in the form: 
+``VERSION-prerelease+build``.
+Conan will also order pre-releases and builds according to the same rules, and each one of them can also
+contain an arbitrary number of items, like ``1.2.3-pre.1.2.1+build.45.a``.
+Note that the semver standard do not apply any ordering to builds.
```

Comment:
```suggestion
Note that the semver standard does not apply any ordering to builds.
```
Also, this is what semver oficially says. It would be nice to clarify what Conan actually ends up doing